crypto/README.md

```diff
@@ -34,16 +44,34 @@ The parameters used to instantiate the functions are:
+* For instantiation `GriffinJive64_256`, we also make the following modifications:
+  - Instead of using the matrix suggested by the Griffin specification paper, we use a methodology developed by Polygon Zero to find an MDS matrix with coefficients which are small powers of two in frequency domain. This allows us to dramatically reduce MDS matrix multiplication time. Using a different MDS matrix does not affect security of the hash function as any MDS matrix satisfies Griffin construction (as described in section 4.2 of the paper).
+  - We use Jive as compression mode for 2-to-1 hashing. Similarly to `RpJive64_256` above, the output of the hash function is not the same when we hash 8 field elements as a sequence of elements using `hash_elements()` function and when we compress 8 field elements into 4 (e.g., for building a Merkle tree) using the 2-to-1 Jive compression mode.
+
+The parameters used to instantiate the function are:
+* For `GriffinJive64_256`:
+  - Field: 64-bit prime field with modulus 2<sup>64</sup> - 2<sup>32</sup> + 1.
+  - State width: 8 field elements.
+  - Capacity size: 4 field elements.
+  - Digest size: 4 field elements (can be serialized into 32 bytes).
+  - Number of founds: 7.
+  - S-Box degree: 7.
+  - Target security level: 128-bits.
+
 ### Hash function performance
 One of the core operations performed during STARK proof generation is construction of Merkle trees. We care greatly about building these trees as quickly as possible, and thus, for the purposes of STARK protocol, 2-to-1 hash operation (e.g., computing a hash of two 32-byte values) is especially important. The table below contains rough benchmarks for computing a 2-to-1 hash for all currently implemented hash functions.
 
-| CPU                         | BLAKE3_256 | SHA3_256 | RP64_256 | RP62_248 |
-| --------------------------- | :--------: | :------: | :------: | :------: |
-| Apple M1 Pro                | 76 ns      | 227 ns   | 5.1 us   | 7.1 us   |
-| AMD Ryzen 9 5950X @ 3.4 GHz | 62 ns      | 310 ns   | 5.2 us   | 6.9 us   |
-| Core i9-9980KH @ 2.4 GHz    | 66 ns      | 400 ns   | -        | 6.6 us   |
-| Core i5-7300U @ 2.6 GHz     | 81 ns      | 540 ns   | -        | 9.5 us   |
-| Core i5-4300U @ 1.9 GHz     | 106 ns     | 675 ns   | -        | 13.9 us  |
+| CPU                         | BLAKE3_256 | SHA3_256 | RP64_256 | RPJ64_256 | RP62_248 | GriffinJ64_256 |
+| --------------------------- | :--------: | :------: | :------: | :-------: | :------: | :------------: |
+| Apple M1 Pro                | 76 ns      | 227 ns   | 5.1 us   | TODO us   | 7.1 us   | TODO us        |
```

Comment:
On an M1, I'm getting the following results:

- RPJ64_256: 3.8 us
- GriffinJ64_256: 2.2 us

Reply:
Thanks, will update it!